Quorum/apis/block_explorers/source_code.py

```diff
@@ -21,60 +27,36 @@ def _parse_source_code(self) -> None:
         Parses the Solidity source code and stores the contract's AST object.
         """
         source_code_str = "\n".join(self.file_content)
+        tmp_path = Path("tmp.sol")
+
+        with open(tmp_path, 'w') as file:
+            file.write(source_code_str)
 
         try:
-            ast = parser.parse(source_code_str)
-            ast_obj = parser.objectify(ast)
-            contract_name = ast_obj._current_contract.name
-            self._parsed_contract = ast_obj.contracts[contract_name]
+            contract_ast = solcx.compile_files(tmp_path, output_values=["ast"], stop_after="parsing")
+            contract_name = list(contract_ast.keys())[0]
+            self._parsed_contract = contract_ast[contract_name]['ast']
+
         except Exception as e:
             pp.pretty_print(f"Error parsing source code for {self.file_name}: {e}\n"
                             f"Some of the checks will not apply to this contract!!!",
                             pp.Colors.FAILURE)
-
-    def get_constructor(self) -> dict | None:
-        """
-        Retrieves the constructor information from the Solidity contract.
-
-        Returns:
-            (dict | None): Constructor information or None if not found.
-        """
-        if self._parsed_contract:
-            return self._parsed_contract.constructor
-        return None
-
-    def get_dependencies(self) -> list | None:
-        """
-        Retrieves the dependencies from the Solidity contract.
-
-        Returns:
-            (list | None): List of dependencies or None if not found.
-        """
-        if self._parsed_contract:
-            return self._parsed_contract.dependencies
-        return None
-
-    def get_enums(self) -> list | None:
-        """
-        Retrieves the enums from the Solidity contract.
-
-        Returns:
-            (list | None): List of enums or None if not found.
-        """
-        if self._parsed_contract:
-            return self._parsed_contract.enums
-        return None
-
-    def get_events(self) -> list | None:
-        """
-        Retrieves the events from the Solidity contract.
-
-        Returns:
-            (list | None): List of events or None if not found.
-        """
-        if self._parsed_contract:
-            return self._parsed_contract.events
-        return None
+        finally:
+            tmp_path.unlink()
+    
+    def __extract_nodes(self, ast: dict, node_type: ExtractASTOption) -> list:
+        functions = {}
```

Comment:
Just for readability I think we should rename it to ast_option or something like this

Reply:
done

Reply:
are you sure?

Reply:
Yep ExtractASTOption => ASTOption

Reply:
I meant the line i was commenting on...
functions = {} => ast_option = {} since its not only functions, its also variables

Reply:
done